http-server/src/main/java/org/triplea/modules/game/ConnectivityController.java

```diff
@@ -40,9 +37,15 @@ public static ConnectivityController build(final GameListing gameListing) {
       keys = {KeyPart.IP},
       rates = {@Rate(limit = 10, duration = 1, timeUnit = TimeUnit.MINUTES)})
   @RolesAllowed(UserRole.HOST)
-  public boolean checkConnectivity(
+  public Response checkConnectivity(
       @Auth final AuthenticatedUser authenticatedUser, final String gameId) {
-    return connectivityCheck.test(
-        gameListing.getHostForGame(authenticatedUser.getApiKey(), gameId));
+
+    if (!connectivityCheck.gameExists(authenticatedUser.getApiKey(), gameId)) {
+      return Response.status(400).build();
+    }
+
+    return Response.ok()
+        .entity(connectivityCheck.canDoReverseConnect(authenticatedUser.getApiKey(), gameId))
+        .build();
```

Comment:
This is something I have absolutely no experience with:
Basically the result of canDoReverseConnect is false whenever the game does not exist, but that means nothing.

I think I would've coded this in a way so canDoReverseConnect throws an exception whenever the game doesn't exist and map the exception to a http status code somehow.
Is this something Dropwizard is designed to do, or do you think it's better the way it currently is?

Reply:
There are exception mappers:

- https://github.com/triplea-game/triplea/blob/master/http-server/src/main/java/org/triplea/http/IllegalArgumentMapper.java
- https://github.com/triplea-game/triplea/blob/85825be5a9a6ac937fc43d599e2a53a432107f56/http-server/src/main/java/org/triplea/http/ServerApplication.java#L194

Some caution needs to be taken with them as they can go bad if too heavily used. Essentially control-flow-by-exception-handling anti-pattern. Signalling error cases directly via response objects and letting the controller layer or service/module layer decide how to handle is generally more direct (after-all, an exception is a glorified go-to, for better or worse).

The game-id not being there is quite a problem. We would suspect it's a client error, hence 4xx is reasonable. We would not want users to be able to craft a curl command to generate server side 5xx's on demand. 

`canDoReverseConnect`  not throwing is kinda useful as we allow its caller to decide if the situation is an error or not. It's the caller that has more context about whether the game-id is valid to begin with. It's a bit odd though as we ask the same module twice "is this valid", then "try this". It would probably be a bit more appropriate to have a service back the controller, have the server use a 'canDoReverseConnect' function and pass the function the game listing to use. I think that is probably getting to be a bit too over-engineered though.

Reply:
>  or do you think it's better the way it currently is?

I was typing a direct response to this and considered return a flag more deeply, namely "Signalling error cases directly via response objects ". It turns out that becomes cleaner and resolve my concern where we are being very procedural with connectivity check, ie:
```
connectivity check, can you do this?
if yes, connectivity check, please try X
```
Rather than:
```
connectivity check, do X
```

The update is I think a bit cleaner in the end: 1f624bad5